src/object/mod.rs

```diff
@@ -188,6 +188,20 @@ pub trait Object: Value {
         })
     }
 
+    #[cfg(feature = "napi-8")]
+    fn freeze<'a, C: Context<'a>>(&self, cx: &mut C) -> bool {
```

Comment:
I think we can have a better return type on failure (e.g., when trying to freeze a Uint8Array). What do you think about following the [ECMAScript spec](https://262.ecma-international.org/12.0/#sec-object.freeze) and throwing a TypeError?

What do you think of `JsResult<&self>`? Returning `self` would also allow chaining:

```rust
let o = cx.empty_object()
    .freeze(&mut cx)?
    .root(&mut cx);
```

Reply:
Would you be okay with returning a `NeonResult<&Self>`? `Object` does not implement `Managed`, which is required for `JsResult` to be implemented properly. Also, I think there could be some advantage to returning an owned `Self` instead of a reference to it, since the freeze would consume it and then output it again.

Reply:
`NeonResult` should be fine. I don't have a strong sense of whether the chaining is valuable.

I do think it's a little more ergonomic to return a `TypeError` on failure instead of a `bool`--which matches the ECMAScript spec.